rasa/shared/core/slots.py

```diff
@@ -45,9 +47,19 @@ def __init__(
                 influence the predictions of the dialogue polices.
         """
         self.name = name
+        self.mappings = mappings
         self._value = initial_value
         self.initial_value = initial_value
         self._value_reset_delay = value_reset_delay
+        if auto_fill:
+            rasa.shared.utils.io.raise_deprecation_warning(
+                "Slot auto-fill has been deactivated in 3.0 and replaced by a"
+                " new explicit mechanism to specify slot mappings. "
+                "Please refer to the docs to learn more.",
+                "4.0.0",
+                docs=None,
+            )
+            auto_fill = False
```

Comment:
Is this really a deprecation warning if we are just setting it to `False` whenever it is `True`?

Reply:
At the moment because devs expect `auto-fill` to be by default `True` (therefore, not required to explicitly add `auto-fill: true` in the domain), the deprecation warning won't be triggered; it would only be raised only if someone would set it to `True` in the domain. @usc-m [suggested ](https://github.com/RasaHQ/rasa/pull/9644#discussion_r711117715)to keep the default auto-fill to be True as it used to be and raise the deprecation warning. WDYT?

Reply:
I've actioned this so that auto-fill remains `True` by default and raises deprecation warning, however, it makes some tests to fail because it increase the overall number of warnings. Ok to keep it like this and edit the tests with explanatory comments where necessary?

Reply:
Example CI run here: https://github.com/RasaHQ/rasa/pull/9644/checks?check_run_id=3652137803